components/script/dom/svgrenderingcontext.rs

```diff
@@ -0,0 +1,109 @@
+use canvas_traits::webgl::{webgl_channel, WebGLMsgSender, WebGLContextShareMode, GLContextAttributes, WebGLMsg, WebGLVersion};
+use std::cell::Cell;
+use crate::dom::window::Window;
+use euclid::Size2D;
+use crate::dom::bindings::root::{DomRoot, Dom};
+use crate::dom::event::{EventBubbles, EventCancelable, Event};
+use crate::dom::webglcontextevent::WebGLContextEvent;
+use crate::dom::svgsvgelement::SVGSVGElement;
+use dom_struct::dom_struct;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::DomObject;
+use crate::dom::bindings::reflector::MutDomObject;
+use crate::dom::bindings::reflector::Reflector;
+
+#[dom_struct]
+pub struct SVGRenderingContext{
+    reflector_: Reflector,
+    #[ignore_malloc_size_of = "Channels are hard"]
+    webgl_sender: WebGLMsgSender,
+    share_mode: WebGLContextShareMode,
+    #[ignore_malloc_size_of = "Channels are hard"]
+    webrender_image: Cell<Option<webrender_api::ImageKey>>,
+    svg: Dom<SVGSVGElement>,
+}
+
+impl SVGRenderingContext{
+    pub fn new_inherited(
+        window: &Window,
+        size: Size2D<u32>,
+        svg: &SVGSVGElement,
+    ) -> Result<SVGRenderingContext, String>{
+        let webgl_chan = match window.webgl_chan() {
+            Some(chan) => chan,
+            None => panic!("Crash the system!"),
+        };
+
+        let (sender, receiver) = webgl_channel()
+            .unwrap();
+        let attrs = GLContextAttributes{
+            depth: false,
+            stencil: false,
+            alpha: true,
+            antialias: true,
+            premultiplied_alpha: true,
+            preserve_drawing_buffer: false,
+        };
+        webgl_chan
+            .send(WebGLMsg::CreateContext(WebGLVersion::WebGL1, size, attrs, sender))
+            .unwrap();
+        let result = receiver.recv().unwrap();
+
+        result.map(|ctx_data| {
+            let webgl_sender = ctx_data.sender;
+            let share_mode = ctx_data.share_mode;
+            SVGRenderingContext{
+                reflector_: Reflector::new(),
+                webgl_sender,
+                share_mode,
+                webrender_image: Cell::new(None),
+                svg: Dom::from_ref(svg),
+            }
+        })
+    }
+    #[allow(unrooted_must_root)]
+    pub fn new(
+        window: &Window,
+        size: Size2D<u32>,
+        svg: &SVGSVGElement,
+    ) -> Option<DomRoot<SVGRenderingContext>>{
+        match SVGRenderingContext::new_inherited(window, size, svg) {
+            Ok(ctx) => Some(DomRoot::from_ref(&ctx)),
```

Comment:
This won't do what you want because SVGRenderingContext is not a WebIDL-backed object, so we'll probably end up with unexpected crashes when code tries to use it like one. I don't see any reason why SVGRenderingContext is necessary for these changes, but if you want to keep it we need to:
* remove the Reflector member
* store `DomRefCell<Option<SVGRenderingContext>>` inside SVGSVGElement

I think it would be more clear to move all of the data and code out of SVGRenderingContext into SVGSVGElement instead.